### PR TITLE
Skin automation hook

### PR DESCRIPTION
**Pull request by sgupta3112**
Hi @rcvrgs,

I have created new PR for skinAutomationHook branch. Please have a look here.

### CODE REVIEW

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m [hunk 1]
```diff
@@ -63,6 +68,16 @@ - (void)viewDidLoad {
...
                                              object:self.skinController];
+  [[NSNotificationCenter defaultCenter] addObserver: self
+                                           selector:@selector(notificationHandler:)
+                                               name:nil
+                                             object:ooyalaPlayer];
+  
+  // In QA Mode , making textView visible
+  if(self.qaModeEnabled==YES){
+    self.textView.hidden = NO;
+
```
Comment (dchauOoyala) [inline OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m]: Can we remove the extra new line here?

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m [hunk 2]
```diff
@@ -63,6 +68,16 @@ - (void)viewDidLoad {
                                            selector:@selector(notificationHandler:)
                                                name:nil
                                              object:self.skinController];
+  [[NSNotificationCenter defaultCenter] addObserver: self
+                                           selector:@selector(notificationHandler:)
+                                               name:nil
+                                             object:ooyalaPlayer];
+  
+  // In QA Mode , making textView visible
+  if(self.qaModeEnabled==YES){
```
Comment (dchauOoyala) [inline OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m]: Nitpicking but can you add a space after the `if` and around the `==` operator?
`if (self.qaModeenabled == YES)`

Comment (dchauOoyala) [reply OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m]: Also, is using `YES` and `NO` preferred over `true` and `false`?

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/FreewheelPlayerViewController.m [hunk 3]
```diff
@@ -83,32 +93,41 @@ - (void)viewDidLoad {
...
+  
+  NSString *message = [NSString stringWithFormat:@"Notification Received: %@. state: %@. playhead: %f count: %d",
+                       [notification name],
+                       [OOOoyalaPlayer playerStateToString:[self.skinController.player state]],
+                       [self.skinController.player playheadTime], appDel.count];
+  
+  NSLog(@"%@",message);
+  
+  //In QA Mode , adding notifications to the TextView
+  if(self.qaModeEnabled==YES) {
```
Comment (dchauOoyala) [inline OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/FreewheelPlayerViewController.m]: Same comment here as above

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/IMAPlayerViewController.m [hunk 4]
```diff
@@ -55,43 +59,46 @@ - (void)viewDidLoad {
...
-    NSLog(@"%@", message);
+  
+  NSString *message = [NSString stringWithFormat:@"Notification Received: %@. state: %@. playhead: %f count: %d",
+                       [notification name],
+                       [OOOoyalaPlayer playerStateToString:[self.skinController.player state]],
+                       [self.skinController.player playheadTime], appDel.count];
+  NSLog(@"%@",message);
+  
+  //In QA Mode , adding notifications to the TextView
+  if(self.qaModeEnabled==YES) {
```
Comment (dchauOoyala) [inline OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/IMAPlayerViewController.m]: Same comment here as above

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m (hunk 2 above)
Comment (brol1dev) [reply OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m]: @dchauOoyala, to your last question, `YES` and `NO` are preferred over `true` and `false` in ObjC. You'll see it like that everywhere, at the end they're aliases to `true` and `false`.

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m (hunk 1 above)
Comment (sgupta3112) [reply OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/DefaultSkinPlayerViewController.m]: Changes Done..

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/IMAPlayerViewController.m (hunk 4 above)
Comment (sgupta3112) [reply OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/IMAPlayerViewController.m]: changes done

File: OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/FreewheelPlayerViewController.m (hunk 3 above)
Comment (sgupta3112) [reply OoyalaSkinSampleApp/OoyalaSkinSampleApp/Players/FreewheelPlayerViewController.m]: Changes Done

**Review by brol1dev — APPROVED** [review]
I'm approving it, although we need to refactor automation to make it better, and not only automation but the sample apps need a refresh.

**Pull request merged**